> /home/travis/build/zsoltk/paperwork/paperwork-integration/build.gradle:79: Error: A newer version of com.android.support:appcompat-v7 than 23.1.1 is available: 23.2.0 [GradleDependency]
>     compile 'com.android.support:appcompat-v7:23.1.1'
>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /home/travis/build/zsoltk/paperwork/paperwork-integration/build.gradle:83: Error: A newer version of com.android.support:support-annotations than 23.1.1 is available: 23.2.0 [GradleDependency]
>     androidTestCompile "com.android.support:support-annotations:23.1.1"
>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>    Explanation for issues of type "GradleDependency":
>    This detector looks for usages of libraries where the version you are using
>    is not the current stable release. Using older versions is fine, and there
>    are cases where you deliberately want to stick with an older version.
>    However, you may simply not be aware that a more recent version is
>    available, and that is what this lint check helps find.
> 2 errors, 0 warnings
> Wrote HTML report to file:///home/travis/build/zsoltk/paperwork/paperwork-integration/build/outputs/lint-results.html
> Wrote XML report to file:///home/travis/build/zsoltk/paperwork/paperwork-integration/build/outputs/lint-results.xml
> :paperwork-integration:lint FAILED
> FAILURE: Build failed with an exception.
> - What went wrong:
>   Execution failed for task ':paperwork-integration:lint'.
>   Lint found errors in the project; aborting build.
>   Fix the issues identified by lint, or add the following to your build script to proceed with errors:
>   ...
>   android {
>       lintOptions {
>           abortOnError false
>       }
>   }
>   ...
> - Try:
>   Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
>   BUILD FAILED
>   Total time: 1 mins 31.648 secs

--- a/paperwork-integration/src/androidTest/java/hu/supercluster/paperwork/integration/test/MainActivityTest.java
+++ b/paperwork-integration/src/androidTest/java/hu/supercluster/paperwork/integration/test/MainActivityTest.java
@@ -4,6 +4,7 @@ import android.support.test.espresso.matcher.ViewMatchers;
 import android.support.test.rule.ActivityTestRule;
 import android.support.test.runner.AndroidJUnit4;
 
+import org.hamcrest.Matchers;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TestRule;
@@ -21,6 +22,8 @@ import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
 import static org.hamcrest.CoreMatchers.any;
 import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.isEmptyString;
+import static org.hamcrest.core.IsNot.not;
 
 @RunWith(AndroidJUnit4.class)
 public class MainActivityTest {
@@ -44,17 +47,17 @@ public class MainActivityTest {
 
     @Test
     public void shouldDisplayGitTag() {
-        onView(ViewMatchers.withId(hu.supercluster.paperwork.integration.R.id.gitTag)).check(matches(withText(any(String.class))));
+        onView(ViewMatchers.withId(hu.supercluster.paperwork.integration.R.id.gitTag)).check(matches(not(withText(isEmptyString()))));
     }
 
     @Test
     public void shouldDisplayGitInfo() {
-        onView(ViewMatchers.withId(hu.supercluster.paperwork.integration.R.id.gitInfo)).check(matches(withText(any(String.class))));
+        onView(ViewMatchers.withId(hu.supercluster.paperwork.integration.R.id.gitInfo)).check(matches(not(withText(isEmptyString()))));
     }
 
     @Test
     public void shouldDisplayGitBranch() {
-        onView(ViewMatchers.withId(hu.supercluster.paperwork.integration.R.id.gitBranch)).check(matches(withText(any(String.class))));
+        onView(ViewMatchers.withId(hu.supercluster.paperwork.integration.R.id.gitBranch)).check(matches(not(withText(isEmptyString()))));
     }
 
     @Test
@@ -84,6 +87,6 @@ public class MainActivityTest {
 
     @Test
     public void shouldDisplayEnv() {
-        onView(ViewMatchers.withId(hu.supercluster.paperwork.integration.R.id.env)).check(matches(withText(any(String.class))));
+        onView(ViewMatchers.withId(hu.supercluster.paperwork.integration.R.id.env)).check(matches(not(withText(isEmptyString()))));
     }
 }

--- a/paperwork-plugin/src/main/groovy/hu/supercluster/paperwork/plugin/PaperworkPluginExtension.groovy
+++ b/paperwork-plugin/src/main/groovy/hu/supercluster/paperwork/plugin/PaperworkPluginExtension.groovy
@@ -26,7 +26,7 @@ class PaperworkPluginExtension {
     }
 
     public String gitBranch() {
-        shell('git branch | sed -n "/\\* /s///p"')
+        ['sh', '-c', 'git branch | sed -n "/\\* /s///p"'].execute().text.trim()
     }
 
     public String buildTime() {

